File: java/src/com/esielkar/facade/Facade.java
package com.esielkar.facade;

public class Facade {
    private Subsystem subsystem = new Subsystem();
    private Subsystem2 subsystem2 = new Subsystem2();
    private AdditonalFacade additonalFacade = new AdditonalFacade();

    public void subsystemsOperation() {
        subsystem.execute();
        subsystem2.execute();
        additonalFacade.subsystemsOperation();
    }
}

File: dart/design_patterns/bin/facade/facade.dart
import 'subsystem.dart';

class Facade {
  final _subsystem = new Subsystem1();
  final _subsystem2 = new Subsystem2();
  final _additonalFacade = new AdditonalFacade();

  void subsystemsOperation() {
    _subsystem.execute();
    _subsystem2.execute();
    _additonalFacade.subsystemsOperation();
  }
}

class AdditonalFacade {
  final _subsystem3 = Subsystem3();

  void subsystemsOperation() {
    _subsystem3.execute();
  }
}
